dyn/resource_dyn_record_test.go

```diff
@@ -163,6 +163,54 @@ func TestAccDynRecord_CNAME_trailingDot(t *testing.T) {
 	})
 }
 
+func TestAccDynRecord_NS_record(t *testing.T) {
+	var record dynect.Record
+	zone := os.Getenv("DYN_ZONE")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDynRecordDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccCheckDynRecordConfig_NS_record, zone),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDynRecordExists("dyn_record.foobar", &record),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "name", "dev"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "type", "NS"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "ttl", "3600"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "zone", zone),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "value", "ns.terraform.io."),
```

Comment:
any reason why this isn't `ns.hashicorptest.com`?

Reply:
so it can point to anything and it wouldn't make too much sense for an NS record to point back to the same domain - Ain't saying it's totally unrealistic, but unlikely when you're delegating.